src/EIPScanner/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# Add library target for common source files
+set(SOURCE_FILES 
+        BaseObject.cpp
+        ConnectionManager.cpp
+        DiscoveryManager.cpp
+        FileObject.cpp
+        IdentityObject.cpp
+        IOConnection.cpp
+        MessageRouter.cpp
+        ParameterObject.cpp
+        SessionInfo.cpp)
+        
+add_library(EIPScanner SHARED ${SOURCE_FILES})
+add_library(EIPScannerS STATIC ${SOURCE_FILES})
+
+
+add_subdirectory(cip)
+add_subdirectory(eip)
+add_subdirectory(fileObject)
+add_subdirectory(sockets)
+add_subdirectory(utils)
```

Comment:
Add a CMakeLists.txt file to every sub_directory to clean up pathing.

Each subdirectory is build independently and then linked back to the parent library

Reply:
** Added a source to the parent library.

I found a great source by a maintainer of CMake and updated all of the CMakeLists.txt files to use the `target_sources()` function.

https://crascit.com/2016/01/31/enhanced-source-file-handling-with-target_sources/